test/es-module/test-typescript.mjs

```diff
@@ -301,26 +315,66 @@ test('execute a TypeScript file with CommonJS syntax requiring .mts with require
 });
 
 test('execute a TypeScript file with CommonJS syntax requiring .mts with require-module with default-type commonjs',
-     async () => {
-       const result = await spawnPromisified(process.execPath, [
-         '--experimental-strip-types',
-         '--experimental-default-type=commonjs',
-         '--no-warnings',
-         fixtures.path('typescript/ts/test-require-cts.ts'),
-       ]);
-
-       strictEqual(result.stderr, '');
-       match(result.stdout, /Hello, TypeScript!/);
-       strictEqual(result.code, 0);
-     });
-
-test('execute a JavaScript file importing a cjs TypeScript file', async () => {
+  async () => {
   const result = await spawnPromisified(process.execPath, [
     '--experimental-strip-types',
+    '--experimental-default-type=commonjs',
     '--no-warnings',
-    fixtures.path('typescript/ts/issue-54457.mjs'),
+    fixtures.path('typescript/ts/test-require-cts.ts'),
   ]);
+
   strictEqual(result.stderr, '');
   match(result.stdout, /Hello, TypeScript!/);
   strictEqual(result.code, 0);
 });
+
+test('execute a TypeScript file with Partial Union Type', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    '--no-warnings',
+    fixtures.path('typescript/ts/test-partial-union-type.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /Hello, TypeScript!/);
+  strictEqual(result.code, 0);
+});
+
+test('execute a TypeScript file with Extract Union Type', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    '--no-warnings',
+    fixtures.path('typescript/ts/test-extract-union-type.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /admin/);
+  strictEqual(result.code, 0);
+});
+
+test('execute a TypeScript file with Union Types', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    '--no-warnings',
+    fixtures.path('typescript/ts/test-union-types.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /Hello, TypeScript!/);
+  strictEqual(result.code, 0);
+});
+
+test('expect error when executing a TypeScript file with generics', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    fixtures.path('typescript/ts/test-generics.ts'),
```

Comment:
if we are testing paramenter properties, why call it `test-generics`?

Reply:
I call `test-generics` because I created the Typescript generics fixture. Do you have any suggestions for a name for the file?

Reply:
`test-parameter-properties`?

Reply:
good! Thanks! I will change the file name